src/com/sap/piper/tools/JavaArchiveDescriptor.groovy

```diff
@@ -11,30 +11,59 @@ class JavaArchiveDescriptor extends ToolDescriptor {
     final javaTool
     final javaOptions
 
-    JavaArchiveDescriptor(name, environmentKey, stepConfigurationKey, executablePath, executableName, version, versionOption, javaTool, javaOptions = '') {
-        super(name, environmentKey, stepConfigurationKey, executablePath, executableName, version, versionOption)
+    JavaArchiveDescriptor(name, environmentKey, stepConfigurationKey, version, versionOption, javaTool, javaOptions = '') {
+        super(name, environmentKey, stepConfigurationKey, '', '', version, versionOption)
         this.javaTool = javaTool
         this.javaOptions = javaOptions
     }
 
     @Override
     def getToolLocation(script, configuration, log = true) {
 
-        def home
+        def javaArchiveLocation
+        def javaArchivePath
         if (EnvironmentUtils.isEnvironmentVariable(script, environmentKey)) {
-            home = EnvironmentUtils.getEnvironmentVariable(script, environmentKey)
-            if (log) script.echo "$name home '$home' retrieved from environment."
+            javaArchiveLocation = EnvironmentUtils.getEnvironmentVariable(script, environmentKey)
+            if (log) script.echo "$name location '$javaArchiveLocation' retrieved from environment."
+            javaArchivePath = getJavaArchivePath(javaArchiveLocation)
+            if (javaArchivePath == null) script.error "$environmentKey has an unexpected format."
         }
         else if (configuration.containsKey(stepConfigurationKey)) {
-            home = configuration.get(stepConfigurationKey)
-            if (log) script.echo "$name home '$home' retrieved from configuration."
+            javaArchiveLocation = configuration.get(stepConfigurationKey)
+            if (log) script.echo "$name location '$javaArchiveLocation' retrieved from configuration."
+            javaArchivePath = getJavaArchivePath(javaArchiveLocation)
+            if (javaArchivePath == null) script.error "$stepConfigurationKey has an unexpected format."
```

Comment:
~~has an unexpected format~~. Is null.

Reply:
It is not null, it has an unexpected format and `getJavaArchivePath` returns null in that case.

Reply:
understood, I read it wrong ...